posthog/hogql/printer.py

```diff
@@ -272,7 +272,7 @@ def visit_join_expr(self, node: ast.JoinExpr) -> JoinExprResponse:
                     cte = f"{self._print_identifier(node.alias)} AS (SELECT * FROM {sql})"
 
                     # The table is captured in a CTE so just print the table name in the final select
-                    sql = self._print_identifier(table_type.table.name)
+                    sql = self._print_identifier(node.alias or table_type.table.name)
```

Comment:
Forgot to add alias as a possible identifier here if it exists

Reply:
Hmm, actually it seems like node.alias is populated even with the table name so table_type.table.name is never needed here even though possible None type? I think node.alias is always given the table_name by default

Reply:
The alias is added a few lines lower:
```py
            if isinstance(node.type, ast.TableAliasType) and node.alias is not None and node.alias != sql:
                join_strings.append(f"AS {self._print_identifier(node.alias)}")
```

things look ok to me... what was the case that was broken? 🤔